CMakeLists.txt

```diff
@@ -4,186 +4,233 @@
+  test/testDerived.cxx
+  test/testFlume.cxx
+  test/testMetric.cxx
+  test/testProcessDetails.cxx
+  test/testProcessMonitor.cxx
+  test/testInfluxDb.cxx
+  test/testNoop.cxx
+)
+
+if(ApMon_FOUND)
   list(APPEND TEST_SRCS test/testApMon.cxx)
-  configure_file(test/ApMon.conf ApMon.conf COPYONLY)
+  configure_file(test/ApMon.conf ${CMAKE_CURRENT_BINARY_DIR}/ApMon.conf COPYONLY)
 endif()
 
-O2_GENERATE_TESTS(
-        MODULE_LIBRARY_NAME ${LIBRARY_NAME}
-        BUCKET_NAME ${BUCKET_NAME}
-        TEST_SRCS ${TEST_SRCS}
-)
+foreach (test ${TEST_SRCS})
+  get_filename_component(test_name ${test} NAME)
+  string(REGEX REPLACE ".cxx" "" test_name ${test_name})
+
+  add_executable(${test_name} ${test})
+  target_link_libraries(${test_name}
+    PRIVATE 
+      Monitoring Boost::unit_test_framework Boost::filesystem
+  )
+  add_test(NAME ${test_name} COMMAND ${test_name})
+  set_tests_properties(${test_name} PROPERTIES TIMEOUT 60)
+endforeach()
 
-set(EXAMPLES
-        examples/1-Basic.cxx
-        examples/2-TaggedMetrics.cxx
-        examples/3-UserDefinedTimestamp
-        examples/4-RateDerivedMetric.cxx
-        examples/5-Benchmark.cxx
-        examples/6-Increment.cxx
-        examples/7-Latency.cxx
-        examples/8-Multiple.cxx
-        examples/9-Timer.cxx
-        examples/10-Buffering.cxx
-	examples/11-AutoUpdate.cxx
-        )
 
-foreach (example ${EXAMPLES})
+####################################
+# Install
+####################################
+
+include(GNUInstallDirs)
+
+# Build targets with install rpath on Mac to dramatically speed up installation
+# https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/RPATH-handling
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  if("${isSystemDir}" STREQUAL "-1")
+    set(CMAKE_INSTALL_RPATH "@loader_path/../lib")
+  endif()
+  set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+endif()
+unset(isSystemDir)
 
-    get_filename_component(example_name ${example} NAME)
-    string(REGEX REPLACE ".cxx" "" example_name ${example_name})
+# Set the default build type to "RelWithDebInfo"
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo"
+    CACHE
+    STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel Coverage."
+    FORCE
+  )
+endif(NOT CMAKE_BUILD_TYPE)
 
-    O2_GENERATE_EXECUTABLE(
-            EXE_NAME ${example_name}
-            SOURCES ${example}
-            MODULE_LIBRARY_NAME ${LIBRARY_NAME}
-            BUCKET_NAME ${BUCKET_NAME}
-    )
+# Install library
+install(TARGETS Monitoring
+  EXPORT monitoring-export
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 
-endforeach()
+# Install headers
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/Monitoring DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+
+# Export targets
+install(EXPORT monitoring-export
```

Comment:
I think that it is often called "MonitoringTargets" instead of "monitoring-export". This is not important though :)